### PR TITLE
Adds CinemaHall, MovieSession, and corresponding DAOs and Services

### PR DESCRIPTION
**Pull request by BusyhinD**

### CODE REVIEW

File: src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            throw new DataProcessingException("Can't get cinema hall by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<CinemaHall> getAllCinemaHallsQuery =
+                    session.createQuery("from CinemaHall", CinemaHall.class);
+            return getAllCinemaHallsQuery.getResultList();
```
Comment (uliana-andrushchuk) [inline src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java:52]: please, add catch block to this method

File: src/main/java/mate/academy/dao/impl/MovieDaoImpl.java [hunk 2]
```diff
@@ -45,6 +46,9 @@ public Optional<Movie> get(Long id) {
 
     @Override
     public List<Movie> getAll() {
-        return null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Movie> allMoviesQuery = session.createQuery("from Movie", Movie.class);
+            return allMoviesQuery.getResultList();
+        }
```
Comment (uliana-andrushchuk) [inline src/main/java/mate/academy/dao/impl/MovieDaoImpl.java:52]: Add `catch` block

File: src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get movie session by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> avaibleSessionsQuery =
+                    session.createQuery("select ms from MovieSession ms "
```
Comment (uliana-andrushchuk) [inline src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java:53]:
```suggestion
                    session.createQuery("SELECT ms FROM MovieSession ms "
```
It's better to use UPPERCASE in queries

File: src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    private MovieSessionDao movieSessionDao;
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        return movieSessionDao.add(movieSession);
+    }
+
+    @Override
+    public MovieSession get(Long id) {
+        return movieSessionDao.get(id).get();
```
Comment (uliana-andrushchuk) [inline src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java:23]: It's better to use `orElseThrow()` method here to throw an exception

**Review by uliana-andrushchuk — CHANGES_REQUESTED** [review]

**Review by kozhukhovsky — APPROVED** [review]

**Pull request closed without merging**